app/assets/javascripts/components/sampleUpload.jsx

```diff
@@ -0,0 +1,550 @@
+class SampleUpload extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.handleUpload = this.handleUpload.bind(this);
+    this.handleUpdate = this.handleUpdate.bind(this);
+    this.csrf = props.csrf;
+    this.project = props.projectInfo ? props.projectInfo : null;
+    this.handleProjectSubmit = this.handleProjectSubmit.bind(this);
+    this.clearError = this.clearError.bind(this);
+    this.handleNameChange = this.handleNameChange.bind(this);
+    this.handleProjectChange = this.handleProjectChange.bind(this);
+    this.handleHostChange = this.handleHostChange.bind(this);
+    this.handleQueueChange = this.handleQueueChange.bind(this);
+    this.handleMemoryChange = this.handleMemoryChange.bind(this);
+    this.handleResultChange = this.handleResultChange.bind(this);
+    this.projects = props.projects || [];
+    this.hostGenomes = props.host_genomes || [];
+    this.hostName = this.hostGenomes.length ? this.hostGenomes[0].name : '';
+    this.hostId = this.hostGenomes.length ? this.hostGenomes[0].id : null;
+    this.sample = props.selectedSample || ''
+    this.selectedSample = {
+      name: this.sample.name || '',
+      hostGenome: this.sample.host_genome_name || '',
+      hostGenomeId: this.sample.host_genome_id || null,
+      project: props.projectInfo ? props.projectInfo : '',
+      resultPath: this.sample.s3_preload_result_path || '',
+      jobQueue: this.sample.job_queue || '',
+      memory: this.sample.sample_memory || '',
+      id: this.sample.id || '',
+      inputFiles: props.inputFiles && props.inputFiles.length ? props.inputFiles : [],
+      projectId: this.project ? this.project.id : null
+    }
+    this.state = {
+      allProjects: this.projects || [],
+      hostGenomes: this.hostGenomes || [],
+      hostName: this.hostName,
+      hostId: this.hostId,
+      invalid: false,
+      errorMessage: '',
+      success: false,
+      successMessage: '',
+      project: 'Select a Project',
+      projectId: null,
+      job_queue: 'aegea_batch',
+      memory: 64000,
```

Comment:
I think the reason why the memory is not set correctly is that this field should be called sample_memory rather than memory.

Reply:
@yunfangjuan, the sample memory is pre-filled if there is sample_memory data, if there isn't, the field is empty and open to edit.

Reply:
@yunfangjuan, You could check a sample that contains sample_memory data, u'll see that the field is pre-filled

Reply:
you can try to create a new sample and see the data in the rails console, the sample_memory is not set there although it was set to 64000